env.go

```diff
@@ -750,11 +750,14 @@ func (e *Env) Exec(l *ui.UI, pkg *manifest.Package, binary string, args []string
 		if filepath.Base(bin) != filepath.Base(binary) {
 			continue
 		}
-		b.Tracef("exec %s", shellquote.Join(append([]string{bin}, args...)...))
+		argsCopy := make([]string, len(args))
+		copy(argsCopy, args)
+		argsCopy[0] = bin
+		b.Tracef("exec %s", shellquote.Join(argsCopy...))
 		l.Clear()
 		timer()
 
-		err = syscall.Exec(bin, args, env)
+		err = syscall.Exec(bin, argsCopy, env)
```

Comment:
Why a copy instead of `args := append(bin, args[1:]...)`

Asking out of curiosity, not as commentary.

Reply:
Because slices are so annoyingly "mutable", I prefer to defensively copy them in case future me decides to do something stupid.

Reply:
Does `append([]string{bin}, args[1:])` copy?

Reply:
No seriously, I don't know!!! 😅

Reply:
Oh, sorry!

That's exactly the issue I have :( and why I do the simplest thing possible mostly, just in [case I get it wrong](https://ueokande.github.io/go-slice-tricks/).

In fact, [here's](https://go.dev/play/p/av5iNsrcBlM) a great example.

Reply:
But in your specific example I think it would be okay because we control the slice being appended to.

Reply:
Is `[]string{bin}` being appended to? And it's a new … slice … yes? 🤷‍♂️